keras_nlp/models/bart/bart_seq_2_seq_lm.py

```diff
@@ -326,22 +326,17 @@ def call_decoder_with_cache(
             )
 
             if self_attention_cache_update_index is not None:
-                self_attention_caches[i] = next_self_attention_cache
+                self_attention_caches.append(next_self_attention_cache)
             if cross_attention_cache_update_index is not None:
-                cross_attention_caches[i] = next_cross_attention_cache
+                cross_attention_caches.append(next_cross_attention_cache)
 
         if self_attention_cache_update_index is not None:
-            self_attention_cache = tf.stack(self_attention_caches, axis=1)
+            self_attention_cache = ops.stack(self_attention_caches, axis=1)
         if cross_attention_cache_update_index is not None:
-            cross_attention_cache = tf.stack(cross_attention_caches, axis=1)
+            cross_attention_cache = ops.stack(cross_attention_caches, axis=1)
 
         hidden_states = x
-
-        logits = tf.matmul(
-            hidden_states,
-            self.backbone.get_layer("token_embedding").embeddings,
-            transpose_b=True,
-        )
+        logits = self.get_layer("reverse_embedding")(x)
```

Comment:
Just curious, as I see this is done throughout -- what's the purpose of using `self.get_layer` instead of storing the layer as a member on the object directly e.g. `self.reverse_embedding`?

Reply:
Because we are functional, they actually aren't `self.layer` or `self.backbone.layer` accessors.